lib/fetching/bot.js

```diff
@@ -34,14 +34,15 @@ util.inherits(Bot, EventEmitter);
 Bot.prototype.start = function() {
   if (this.enabled) return;
   this.enabled = true;
+  this.source.enabled = true;
```

Comment:
The major problem was that the status of the source was not being properly reflected in the bot-master. So, at some point, the source.enabled would be 'true', but the bot would've been killed (by disabling the source on the sources page). In this case, when Fetching was toggled, it somehow used to re-enable a source, despite the sources page saying it was disabled.